src/LinAlg/hiopVectorIntRaja.cpp

```diff
@@ -106,4 +106,13 @@ void hiopVectorIntRaja::copy_to_dev()
   }
 }
 
+void hiopVectorIntRaja::copy_from(const index_type* v_local)
```

Comment:
I can create a copy constructor in classes "hiopVectorIntSeq" and "hiopVectorIntRaja", but probably we still can keep this function since it can update the vector value.
BTW, I changed some other implementations and hence right now this `copy_from` function is not used in HiOp.

Reply:
I think copy constructor would not replace the functionality of the method entirely.  But it would be nice to have it go and do not allow to copy to or from raw pointers

Reply:
Could we document the difference? I'm not sure I can tell from the header what the difference is. Perhaps if we don't want a copy constructor we could delete it so we never misuse it.

Reply:
I think Cosmin agree to have copy constructors in the corresponding classes and we can open another PR to add them.

We may need a function to copy from a raw pointer, since we have `iterate_callback` function which allow user to change the values of `hiopVector` between two iterations. See `hiopVector::copyFrom(double*)` for an example.
Therefore, I follow the same pattern to create this function, i.e., `hiopVectorInt::copy_from(index_type*)`.
This function was used in unit test,  but now I am using new implementations and this one is not used anymore.

Reply:
I checked the code again. Please let me correct my reply above.
`hiopVector::copyFrom(double*)` is NOT used in the `iterate_callback`, but used in some callback functions in Ex1 and primal-decomposition solver. It is also used in some MPI functions. For example, copy back the MPI_All_Reduce results to a vector.

Again, I am fine to delete this function for this PR right now since it is not used. Or just leave it for now and create another issue to address it by a copy constructor. It is not urgent at all, since we may not need this function in the near future.

Reply:
yes, fill an issue for copy constructor (my opinion is that we should not support them). Then let's ensure `copy_from` works fine and try to merge this.

Reply:
@cnpetra, isn't `new_copy` method in `hiopVector` supposed to create a deep copy of `*this`? Perhaps having a similar method/implementation in the `hioVectorInt` would be a consistent solution?

Reply:
yes, `new_copy` and `alloc_clone` are preferred to having copy constructors...